### PR TITLE
Implementing Model Classes and Custom Methods

### PR DESCRIPTION
**Pull request by PabloBona**
**Changes Made:**

1. **Model Classes:** Created model classes for all entities based on the provided Entity-Relationship Diagram (ERD). This includes `User`, `Post`, `Comment`, and `Like` models.

2. **Associations:** Established associations between the models as per the project requirements. Notably, the `author_id` column in the `posts` table is set as a foreign key referencing the `users` table.

3. **Custom Methods:** Added custom methods to our models, enhancing their capabilities:
    - **Users:** Created a method that retrieves the three most recent posts for a given user.
    - **Posts:** Implemented a method for updating the post counter for a user and another method that retrieves the five most recent comments for a post.
    - **Comments:** Added a method for updating the comment counter for a post.
    - **Likes:** Included a method for updating the like counter for a post.
    
image

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: :author_id
+  has_many :comments
+  has_many :likes
```
Comment (topeogunleye) [inline app/models/user.rb]: Hello @PabloBona,

Great job on setting up the associations in your User model! However, I noticed a slight discrepancy between the project requirements and your implementation.

As per the ERD Diagram, the User has many posts, comments, and likes. it is specified that the foreign_key for all the has_many associations should be 'author_id'. In your code, you've correctly specified the foreign_key for has_many :posts, but missed doing so for has_many :comments and has_many :likes.


Remember, you're making good progress!  Keep up the good work!

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @PabloBona, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Try-Again

Highlights 🌟

1. Model Organization: You've neatly organized the User, Post, Comment, and Like models inside the app/models/ directory. Good job! 📁

2. Associations: All associations between your models are set up exactly as per the given ERD diagram. This shows a great understanding of Rails associations. 🤝

3. Methods: Your recent_post method in the User model, recent_comments in the Post model, as well as counter-updating methods in the Post, Comment, and Like models are well-implemented. 👍

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Nothing to mention.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (PabloBona) [reply app/models/user.rb]: Thanks for your feedback.
Changes have already been implemented!

Comment (topeogunleye) [reply app/models/user.rb]: Great. please submit for review

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @PabloBona,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by PabloBona** [comment]
Thanks @Omar-Muhamad  !

File: app/models/user.rb (hunk 1 above)
Comment (PabloBona) [reply app/models/user.rb]: @topeogunleye It wasn't good advice.  Is not author_id is user_id... I have got a lot of issues in spec...

Comment (topeogunleye) [reply app/models/user.rb]: What problems do you have with your specs

**Pull request merged**